### PR TITLE
Engine mutable receiver

### PR DESCRIPTION
**Pull request by aludvik**

### CODE REVIEW

File: sdk/rust/src/consensus/zmq_driver.rs [hunk 1]
```diff
@@ -74,13 +74,14 @@ pub fn start<T: AsRef<str>, E: Engine + Send + 'static >(endpoint: T, engine: E)
 
         let engine_thread = ::std::thread::spawn(move || {
             eprintln!("starting engine");
+            let (name, version) = { (engine.name(), engine.version()) };
             engine.start(
                 update_receiver,
                 Box::new(ZmqService::new(
                     validator_sender_clone,
                     ::std::time::Duration::from_secs(SERVICE_TIMEOUT),
-                    engine.name(),
```
Comment (nick-drozd) [inline sdk/rust/src/consensus/zmq_driver.rs]: Is this a functional change?

Comment (aludvik) [reply sdk/rust/src/consensus/zmq_driver.rs]: In order to borrow the engine as mutable here, it needs to not be borrowed anywhere else.

**Comment by nick-drozd** [comment]
tested locally, works

**Comment by nick-drozd** [comment]
works

**Pull request merged**